Assets/Python/RFCUtils.py

```diff
@@ -677,274 +569,180 @@ def isAVassal(self, iCiv):
-						if (pPlot.countTotalCulture() == 0 ):
-							# this is a good plot, so paint it and continue search
-							return (None, bPaint, bContinue)
-		# not a good plot, so don't paint it but continue search
-		return (None, not bPaint, bContinue)
+		pPlot = gc.getMap().plot(tCoords[0], tCoords[1])
+		if pPlot.isHills() or pPlot.isFlatlands():
+			if not pPlot.isImpassable():
+				if not pPlot.isUnit():
+					if pPlot.getTerrainType() not in [iDesert, iTundra, iMarsh] and pPlot.getFeatureType() != iJungle:
+						if pPlot.countTotalCulture() == 0:
+							return True
+		return False
 
 	#RiseAndFall
-	def cityPlots(self, tCoords, result, argsList):
-		bPaint = True
-		bContinue = True
+	def cityPlots(self, tCoords, argsList):
 		pPlot = gc.getMap().plot(tCoords[0], tCoords[1])
 		if pPlot.isCity():
-			return (None, bPaint, bContinue)
-		return (None, not bPaint, bContinue)
+			return True
+		return False
 
-	def ownedCityPlots( self, tCoords, result, argsList ):
+	def ownedCityPlots(self, tCoords, argsList):
 		"""Checks validity of the plot at the current tCoords, returns plot if valid (which stops the search).
 		Plot is valid if it contains a city belonging to the civ"""
-		bPaint = True
-		bContinue = True
-		pPlot = gc.getMap().plot( tCoords[0], tCoords[1] )
-		if (pPlot.getOwner() == argsList ):
-			if (pPlot.isCity()):
-				# this is a good plot, so paint it and continue search
-				return (None, bPaint, bContinue)
-		# not a good plot, so don't paint it but continue search
-		return (None, not bPaint, bContinue)
-
-	def ownedCityPlotsAdjacentArea( self, tCoords, result, argsList ):
+		pPlot = gc.getMap().plot(tCoords[0], tCoords[1])
+		if pPlot.getOwner() == argsList:
+			if pPlot.isCity():
+				return True
+		return False
+
+	def ownedCityPlotsAdjacentArea(self, tCoords, argsList): # Unused
 		"""Checks validity of the plot at the current tCoords, returns plot if valid (which stops the search).
 		Plot is valid if it contains a city belonging to the civ"""
-		bPaint = True
-		bContinue = True
-		pPlot = gc.getMap().plot( tCoords[0], tCoords[1] )
+		pPlot = gc.getMap().plot(tCoords[0], tCoords[1])
 		#print(tCoords[0], tCoords[1], pPlot.isCity(), pPlot.getOwner() == argsList[0], pPlot.isAdjacentToArea(gc.getMap().plot(argsList[1][0],argsList[1][1]).area()))
-		if (pPlot.getOwner() == argsList[0] and pPlot.isAdjacentToArea(gc.getMap().plot(argsList[1][0],argsList[1][1]).area())):
-			if (pPlot.isCity()):
-				# this is a good plot, so paint it and continue search
-				return (None, bPaint, bContinue)
-		# not a good plot, so don't paint it but continue search
-		return (None, not bPaint, bContinue)
-
-	def foundedCityPlots( self, tCoords, result, argsList ):
+		if pPlot.getOwner() == argsList[0] and pPlot.isAdjacentToArea(gc.getMap().plot(argsList[1][0],argsList[1][1]).area()):

```

Comment:
I'm not really certain what all arguments in the argslist are. I could not find an example, because this function is not used in any python file. (It will be removed soon very likely anyway.)